CANTroller2/src/qpid.h

```diff
@@ -463,15 +465,14 @@ class ThrottleControl {  // Soren - To allow creative control of PID targets in
     uint32_t timestamps_us[20];
     Timer settleTimer, tachHistoryTimer;
   public:
-    ThrottleControl (float* measraw, float* measfilt, float* coolant,  // Variable references: idle target, rpm raw, rpm filt, Engine temp
+    ThrottleControl (float* measraw, float* measfilt,  // Variable references: idle target, rpm raw, rpm filt

```

Comment:
ThrottleControl doesn't take a temperature any more?

Reply:
Throttle was initializing before the engine, so asking for a sensor read before the sensor existed was causing a null pointer crash and reboot loop. The current workaround is to have the constructor for throttle control have an engine temp sensor (pointer) that is initially null, and then there’s a set engine temp sensor method we call during setup() after the sensor exists. 
We could also make the constructors for all of these classes that depend on each other  return pointers instead of instances - apparently in c++ this makes the instances load on use, and we don’t have to worry about the order they initialize and race conditions. 
thoughts?

Reply:
If I understand correctly, the instantiations in globals.h don't necessarily finish in order, but setup() in cantroller should run in order, blocking as needed. I moved the tachometer and temp sensor pointers to a new setup method for ThrottleControl and made sure that both the temp sensors and tachometer run their setup() before the throttle does

Reply:
What would be the benefit of passing the whole sensor manager instead of the sensor we want for the throttle control? I had it before that we create sensors whether they physically exist or not, and there's a has_valid_address method. But I found this was causing more headache than just only creating sensors when they've been detected, and iterating through or getting them as needed.

Reply:
if you passed in the whole manager, then if you got a nullptr for the engine sensor you could always try again later (which would solve your ordering issue). But I like the model where init doesn't have ordering guarantees but setup does, that's very easy to understand and all the ordering logic is right there in one place. so with that in mind, then just passing the sensor works fine.